### PR TITLE
modalselect documented, js library folder, small fixes

### PR DESCRIPTION
**Pull request by robbiejackson**
### **User description**
ModalSelectField documented for 5.2

Also I moved the javascript entries into a subfolder Joomla Javascript library. This is to make way for documents covering how to use js in your extensions, ajax, json response, maybe more.

A couple of fixes in the exampleform component url.

Plus documented icons section
___

### **PR Type**
documentation, enhancement

___

### **Description**
- Added comprehensive documentation for the `ModalSelect` field, including examples and implementation details.
- Corrected URL paths in multiple documentation files to ensure consistency and accuracy.
- Added new documentation sections for Joomla JavaScript Library and icons usage.
- Removed outdated JavaScript overview document.
- Fixed typos and corrected paths to related documentation for better clarity.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>example-form-component.md</strong><dd><code>Correct URL path in example component documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/building-extensions/components/example-form-component.md

<li>Corrected URL path for running the example component.<br> <li> Updated documentation for consistency.<br>

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-43591f08b69ef876f0c1df6a4a973d7d1a77cca75b320297f17a3a077a80aed0">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>step9-ajax.md</strong><dd><code>Update path to core.js documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/building-extensions/modules/module-development-tutorial/step9-ajax.md

- Updated the path to the core.js documentation.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-43c048ca462e1d46fbaeeea8d7250a56e62b4c7f8e73f657ce5dd666b19ce036">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>modalselect.md</strong><dd><code>Add documentation for ModalSelect field</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/forms-fields/standard-fields/modalselect.md

<li>Added comprehensive documentation for the ModalSelect field.<br> <li> Included examples and implementation details.<br>

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-17c4893fd0adbe2c65b933eacc398b645128690334175797c64c58a0b8940336">+221/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>icons.md</strong><dd><code>Add documentation for Joomla icons usage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/icons.md

<li>Added documentation for using icons in Joomla.<br> <li> Explained usage of FontAwesome icons.<br>

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-94f8dc5c44c724024b0a076fda1ba673b1d2b631c095675caa1271a376b0b000">+29/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>Remove outdated JavaScript overview document</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/index.md

- Removed outdated JavaScript overview document.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-e25692ce19116458fe9022e2f600ec448c48cd5d3d13b03374af5514edbe1a50">+0/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>core.md</strong><dd><code>Correct path to WebAssetManager documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/js-library/core.md

- Corrected path to WebAssetManager documentation.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-9315a7ade5625b32a1f1992ae934402ee52159cb87d3ed7db6c557781b55f5ec">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>editors.md</strong><dd><code>Correct path to WebAssetManager documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/js-library/editors.md

- Corrected path to WebAssetManager documentation.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-a2ad5a8ae6f45f7a25af0ba11ee208a015cc755cf01c64244e09aa2126258d6b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>Add Joomla JavaScript Library section</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/js-library/index.md

- Added a new section for Joomla JavaScript Library.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-68272858c5ab1c2a48d1b1d9410934f4ab79780763fd7353362748ab059b5e4e">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>joomla-dialog.md</strong><dd><code>Correct path to WebAssetManager documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/js-library/joomla-dialog.md

- Corrected path to WebAssetManager documentation.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-9aa158c3844230a21684cbcbd5d082fbd33bef1e379f46aa18331ca48e339810">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>modal-content-select.md</strong><dd><code>Fix typo in dialog buttons recommendation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/general-concepts/javascript/js-library/modal-content-select.md

- Fixed a typo in the recommendation for dialog buttons.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-7ae7519c9bb59df4ca7600889381ae2c72f27d66bed23849ebbbb2f6199a5db2">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>example-form-component.md</strong><dd><code>Correct URL path in example component documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

versioned_docs/version-4.4/building-extensions/components/example-form-component.md

- Corrected URL path for running the example component.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-c30cd3c1f6967502ba709f875ce584d4b9e4e33be0708875a0076c8884b03fa6">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>example-form-component.md</strong><dd><code>Correct URL path in example component documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

versioned_docs/version-5.0/building-extensions/components/example-form-component.md

- Corrected URL path for running the example component.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-f68c7b43590667eed760692f86ae6c00a57313cbead8b411948752ced5149a1a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>example-form-component.md</strong><dd><code>Correct URL path in example component documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

versioned_docs/version-5.1/building-extensions/components/example-form-component.md

- Corrected URL path for running the example component.

</details>

  </td>
  <td><a href="https://github.com/joomla/Manual/pull/302/files#diff-8bf5086af1cd28302fb090c5728a8d2c62ae8761cecc5e1bca47350c058fb207">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: docs/general-concepts/forms-fields/standard-fields/modalselect.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  clear="true"
+  sql_title_table="#__content"
+  sql_title_column="title"
+  sql_title_key="id"
+/>
+```
+
+By navigating to "/index.php/component/exampleform/" on your site front-end, this will display a ModalSelect field which will allow you to select an article.
+
+This works because in the site com_content DisplayController constructor it has:
```
Comment (Fedik) [inline docs/general-concepts/forms-fields/standard-fields/modalselect.md:75]: I not very understood this part. What you tries to explain? :wink: 
`ModalSelect` field is "global" field, provided by library, it will work for every part of the site which uses a Form, without extra work.

Comment (Fedik) [reply docs/general-concepts/forms-fields/standard-fields/modalselect.md:75]: Ah I get it, it about the popup content, not about the field.
Maybe better something like this (with better wording):

-------

By navigating to "/index.php/component/exampleform/" on your site front-end, this will display a ModalSelect field which will allow you to select an article.

After clicking the "select" button, the field will open dialog with iframe with content from `urlSelect`, whish is `index.php?option=com_content&view=articles&layout=modal&tmpl=component`, with list of the articles to select. In this example, on the front-end the com_content view will assign the `modal` layout from administrator side, with following code: ....
And on the administrator side com_content will use `modal` layout dircetly,

Comment (robbiejackson) [reply docs/general-concepts/forms-fields/standard-fields/modalselect.md]: Thanks for the suggestions, I've changed the text along the lines you suggested.

I thought it might be best to move the "How it Works" section up as well.

Comment (Fedik) [reply docs/general-concepts/forms-fields/standard-fields/modalselect.md]: Thanks!

> I thought it might be best to move the "How it Works" section up as well.

No need, the example is also part of "how it works", it is good here.
